File: platform/sdk/python/examples/graph/graph-010_spark_writer_bulk.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from xuanji_sdk_graph import GraphClient

client = GraphClient({"endpoint": "graph.local"})
nodes = [{"id": "n1", "label": "User"}, {"id": "n2", "label": "User"}]
edges = [{"id": "e1", "src": "n1", "dst": "n2", "label": "KNOWS"}]
result = client.sparkWriterBulk(nodes, edges)
if not result["ok"] or result["nodesWritten"] != 2 or result["edgesWritten"] != 1:
    sys.exit(1)
print("XJ-OK: graph-010_spark_writer_bulk")
sys.exit(0)

File: platform/sdk/nodejs/examples/graph/graph-010_spark_writer_bulk.js
const { GraphClient } = require("../../mox-sdk-graph");
const client = new GraphClient({ endpoint: "graph.local" });
const nodes = [{ id: "n1", label: "User" }, { id: "n2", label: "User" }];
const edges = [{ id: "e1", src: "n1", dst: "n2", label: "KNOWS" }];
const result = client.sparkWriterBulk(nodes, edges);
if (!result.ok || result.nodesWritten !== 2 || result.edgesWritten !== 1) process.exit(1);
console.log("XJ-OK: graph-010_spark_writer_bulk");
process.exit(0);
